xamarin/Xamarin.Forms
# [iOS] Properly handle EmptyView updates

### Description of Change ###

On iOS, updating the EmptyView while it is visible can cause a crash. And swapping between two EmptyViews at runtime can cause one of the views to remain visible even when there are items in the CollectionView.

These changes address both problems and add a test page for updating EmptyView properties while the EmptyView is visible.

### API Changes ###
 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Open Control Gallery, navigate to CollectionView Gallery -> EmptyView Galleries -> EmptyView (Swap EmptyView). 

Searching for a non-matching term should display the EmptyView; toggling the switch will change the EmptyView property. Switching the EmptyView property should not crash and should display the alternate EmptyView as expected.

### PR Checklist ###
